pyspark/dl/util/common.py

```diff
@@ -135,25 +135,44 @@ def __str__(self):
 
 
 class Sample(object):
-    def __init__(self, features, label, features_shape, label_shape,
-                 bigdl_type="float"):
+    def __init__(self, features, label, bigdl_type="float", features_shape=None, label_shape=None):
         def get_dtype():
             if "float" == bigdl_type:
                 return "float32"
             else:
                 return "float64"
-        self.features = np.array(features, dtype=get_dtype()).reshape(features_shape)  # noqa
-        self.label = np.array(label, dtype=get_dtype()).reshape(label_shape)
+        if features_shape is None:
```

Comment:
Can we remove these changes if those transforming action base on image instead of sample?